silver/models/transactions.py

```diff
@@ -29,3 +39,32 @@ def customer(self):
     @property
     def payment_processor(self):
         return self.payment_method.payment_processor
+
+    @transition(field=status, source='*', target=State.Canceled)
+    def cancel(self):
+        """
+        The transaction is canceled.
+        """
+        self.payment.fail()
+
+    @transition(field=status, source=State.Uninitialized, target=State.Pending,
+                conditions=[lambda t: t.payment.status == t.payment.Status.Unpaid])
+    def pending(self):
+        """
+        Transit to the Pending state.
+        """
+        self.payment.process()
+
+    @transition(field=status, source=State.Pending, target=State.Succeded)
+    def succeed(self):
+        """
+        Finish the transaction successful.
```

Comment:
Maybe something like "The transaction is successful." to be consistent with the other transition descriptions.